### PR TITLE
Add the categorical metadata balancing sampling strategy

### PR DESCRIPTION
**Pull request by mrpositron**
## What has changed and why?

This PR adds a sampling strategy that balances the selection over the values of one categorical metadata key. The strategy works from Python with `multi_strategies([MetadataBalancingStrategy(...)])` and over the REST API with `"strategy_name": "metadata_balance"`. The public `Sampling.metadata_balancing()` method, the docs, and the example are in PR 3 (#2092).

This PR is 2 of 3 and it follows #2090, which is now merged. Of the 677 added lines, 428 are tests.

## How has it been tested?

```bash
cd lightly_studio
uv run pytest tests/sampling/ tests/api/routes/api/test_sampling.py
```

Result: 101 passed. `make static-checks` is clean.

- `tests/sampling/test_metadata_balancing.py` holds 12 tests on `get_metadata_balancing_data`. They cover:
  - the three target distributions and boolean values
  - a target value that no sample has, missing values, and no values at all
  - more distinct values than the strategy balances
  - an explicit target at the limit and above the limit
  - the three error cases

  These are unit tests and not full samplings, because none of them assert anything about the selection.

- `test_sampling_via_db.py` keeps the three cases that need a real sampling: the uniform selection, that a sample without a value stays selectable, and that two keys each add their own balancing. The API route tests cover the `metadata_balance` payload and the rejection of a non-categorical key.

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

The user-facing entry is in PR 3, together with the public method.

## Summary by CodeRabbit

* **New Features**
  * Added metadata-based sampling for categorical values.
  * Supports uniform, input-proportional, and custom target distributions.
  * Handles boolean values, missing metadata, and high-cardinality fields.
  * Supports combining multiple metadata-balancing strategies.

* **Bug Fixes**
  * Added validation for unsupported metadata fields and invalid target distributions.

* **Tests**
  * Added coverage for metadata balancing, edge cases, missing values, and combined strategies.

### CODE REVIEW

File: lightly_studio/src/lightly_studio/sampling/metadata_balancing.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    Values with a target keep it. All other values of the key are grouped as "other"
+    and share the target remaining to 1.0.
+    """
+    target_keys: list[_ColumnKey] = list(target_distribution)
+    target_values = list(target_distribution.values())
+
+    if set(value_counts) - set(target_distribution):
+        target_keys.append(_OTHER)
+        target_values.append(max(1.0 - sum(target_distribution.values()), 0.0))
+    return target_keys, target_values
```
[the remark this answers is not in the document]
Comment (mrpositron) [reply lightly_studio/src/lightly_studio/sampling/metadata_balancing.py]: Duplicate of the issue raised by Codex

**Review by ikondrat — APPROVED** [review]
